apps/system/js/app_chrome.js

```diff
@@ -515,14 +515,19 @@
     }
 
     var self = this;
-    var previousColor;
+    var finishedFade = false;
+    var endBackgroundFade = function() {
+      finishedFade = true;
+      self.element.removeEventListener('transitionend', endBackgroundFade);
+    };
+    this.element.addEventListener('transitionend', endBackgroundFade);
 
     window.requestAnimationFrame(function updateAppColor() {
```

Comment:
Actually, a thought - why is this requestAnimationFrame anyway? If we know what the target colour is, why don't we just calculate if/when the brightness is going to change and use setTimeout? Seems like using requestionAnimationFrame is forcing a lot of unnecessary work to avoid doing some (probably) simple maths...